app/src/main/java/cm/aptoide/pt/home/apps/AppsManager.java

```diff
@@ -276,8 +276,7 @@ public Completable updateApp(App app, boolean isAppcUpdate) {
               return Single.just(value);
             })
             .flatMapCompletable(download -> installManager.install(download))
-            .andThen(aptoideInstallManager.sendConversionEvent()))
-        .toCompletable();
+            .andThen(aptoideInstallManager.sendConversionEvent()));
```

Comment:
@franciscoaleixo i have some doubts here. This chain will get the update from the db in order to create the download. Before, since it was an observable and if the update did nothing would happen. Now since it is a single, it will return null, which might lead to a NP.
Technically, for it to happen would be really weird, because the app would have to be on the list of updates, but the update would not be in the database. I am not seeing how this would happen, so it is a minor problem, but i still think we should try to defend ourselves from these possible mistakes. Currently it will go for the doOnError and log the error. Since it has a retry around the button click, nothing would happen, just like before.